librad/src/identities/git.rs

```diff
@@ -775,11 +775,29 @@ impl<'a> Identities<'a, Project> {
         known: Person,
         latest_head: git2::Oid,
     ) -> Result<VerifiedPerson, error::VerifyPerson> {
+        let is_fast_forward = |known: &Person, head: git2::Oid| {
+            let commit = self.repo.find_commit(head)?;
+            self.is_in_ancestry_path(known.content_id.into(), commit.tree_id())
+        };
         // Nb. technically we could coerce `known` into a `VerifiedPerson` if its
         // `content_id` equals `latest_head`. Let's not introduce an unsafe
         // coercion, but rely on caching to be implemented efficiently.
         if self.is_in_ancestry_path(latest_head, known.revision.into())? {
             self.as_person().verify(latest_head)
+        // If the latest is a fast-forward then we can update the top-level
+        } else if is_fast_forward(&known, latest_head)? {
```

Comment:
I'm not sure about the placement of this, but I was struggling to find where the logic could go. What do you think @kim?

Reply:
Into `replication`. 

I said [here](https://github.com/radicle-dev/radicle-link/pull/711#issuecomment-866714328) that it is probably a mistake to fail verification if _top level_ delegates fail verification. The reason is this:

Imagine you create that project with valid delegates, which are then "checkpointed" as `rad/ids/*`. Now, one of them creates an invalid revision _after_ this checkpoint. If the project fails verification by auto-resolving to the top-level, it can _never_ recover from that (eg. by removing the faulty delegate), because we never replicate it again.

To solve this, we'd need to allow passing the `find_latest_head` function to `git::identities::project::verify`, and make it resolve to `rad/ids/<urn.id>` on the first pass. After replication is done doing its thing, we run verification again, this time resolving to `refs/namespaces/<urn.id>/rad/id`, and return a _different_ value from `replicate` if that fails.